internal/image/ostree_raw.go

```diff
@@ -91,16 +91,27 @@ func (img *OSTreeRawImage) InstantiateManifest(m *manifest.Manifest,
 	buildPipeline.Checkpoint()
 
 	var art *artifact.Artifact
-	switch img.Compression {
-	case "xz":
-		ostreeCompressed := ostreeCompressedImagePipelines(img, m, buildPipeline)
-		art = ostreeCompressed.Export()
-	case "":
-		ostreeBase := baseRawOstreeImage(img, m, buildPipeline)
-		ostreeBase.Filename = img.Filename
-		art = ostreeBase.Export()
+	switch img.Platform.GetImageFormat() {
+	case platform.FORMAT_VMDK:
+		if img.Compression == "" {
+			ostreeBase := baseRawOstreeImage(img, m, buildPipeline)
+			vmdkPipeline := manifest.NewVMDKFromOstreeRaw(m, buildPipeline, ostreeBase)
+			vmdkPipeline.Filename = img.Filename
+			art = vmdkPipeline.Export()
+		}
```

Comment:
The behavior if someone specifies a compression and uses VMDK format will be wrong and won't produce any error, which is not good.

IMO, it would be ideal to refactor the code in some way, so that the compression is applied conditionally regardless of the image format.

But as the bare minimum, I would propose to add an `else` branch with `panic()`.